File: 10474.cpp
#include <array>
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
#include <fstream>

int main(){
    int c = 0;

    while (true){
        int N{};
        int Q{};
        scanf("%d %d", &N, &Q);
        if (N == 0 && Q == 0){
            return 0;
        }
        c++;
        std::vector<int> numbers(N);
        std::vector<int> queries(Q);

        for (int i = 0; i<N; i++){
            scanf("%d", &numbers[i]);
        }
        for (int i = 0; i<Q; i++){
            scanf("%d", &queries[i]);
        }
        std::sort(numbers.begin(), numbers.end());
        std::cout << "CASE# " << c << ":" << std::endl;
        for (int i = 0; i<Q; i++){
            
            if (std::binary_search(numbers.begin(), numbers.end(), queries[i])){
                auto index = std::lower_bound(numbers.begin(), numbers.end(), queries[i]);
                std::cout << queries[i] << " found at " <<(index-numbers.begin()+1) << std::endl;
            }
            else{
                std::cout << queries[i] << " not found" << std::endl;
            }
        }


    
    }

    return 0;
}

File: 10474.py
def main():
    case = 0
    while True:
        N, Q = map(int, input().split())
        if N == 0 and Q == 0:
            return
        else:
            case += 1
        numbers = []
        guesses = []
        for _ in range(N):
            numbers.append(int(input()))
        numbers.sort()
        for _ in range(Q):
            guesses.append(int(input()))
        print("CASE# {}: ".format(case))
        for guess in guesses:
            if guess in numbers:
                print("{} found at {}".format(guess, numbers.index(guess) + 1))
            else:
                print("{} not found".format(guess))


while True:
    try:
        main()
    except EOFError:
        break
